### ISSUE TITLE
Reject all jobs causes batsim to deadlock

### ISSUE THREAD

**Issue by adfaure** [oar-team/batsim#34]
Hello,
I don't know if it matters but if you reject all job batsim will deadlock with:
```
[master_host0:server:(2) 604796.002400] /home/adfaure/Projects/batsim/src/server.cpp:700: [root/CRITICAL] Left simulation loop, but the simulation does NOT seem finished...
Backtrace (displayed in process server):             
---> xbt_backtrace_display_current at ??:?, 0x7f2631d97fd2                                                
---> server_process(int, char**) at /home/adfaure/Projects/batsim/src/server.cpp:88, 0x49c5b2             
---> std::_Function_handler<void (), simgrid::xbt::MainFunction<int (*)(int, char**)> >::_M_invoke(std::_Any_data const&) at ??:?, 0x7f2631d40449                                                                   
---> simgrid::kernel::context::RawContext::wrapper(void*) at ??:?, 0x7f2631d2e512                         
Aborted       
```
Hence the scheduler won't finish and will wait forever.

You can test is using the my scheduler on any workload.
```
cargo run --bin --rej #Will reject all jobs, one by one until the end.
```

**Comment by mpoquet** [comment]
Can reproduce the issue with 7eab359 by running ``cargo run --bin rej``.
Investigating.

**Comment by mpoquet** [comment]
Batsim forgot to check whether the simulation was finished on job rejection.
Fixed in 611e7bf, thanks for the report :).

**Comment by adfaure** [comment]
That was fast ! 
Thanks
